pkg/model/firewall.go

```diff
@@ -408,10 +432,11 @@ func JoinSuffixes(src SecurityGroupInfo, dest SecurityGroupInfo) string {
 func (b *KopsModelContext) AddDirectionalGroupRule(c *fi.ModelBuilderContext, t *awstasks.SecurityGroupRule) {
 
 	name := generateName(t)
+	b.SecurityGroupRules[name] = t
 	t.Name = fi.String(name)
+	t.Delete = fi.Bool(false)
 
-	klog.V(8).Infof("Adding rule %v", name)
-	c.AddTask(t)
+	klog.V(4).Infof("Adding rule %v", name)
```

Comment:
```suggestion
	klog.V(4).Infof("Adding rule %q", name)
```
Consider moving this line up to between lines 434 and 435, once we have the generated name in hand.